cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -300,6 +300,165 @@ class MultivariateAnomalyEstimator(override val uid: String) extends Estimator[D
   }
 }
 
+trait HasBlobHelper extends HasServiceParams {
+
+  val timestampCol = new ServiceParam[String](this, "timestampCol", "Timestamp column name")
+
+  def setTimestampCol(v: String): this.type = setScalarParam(timestampCol, v)
+
+  def getTimestampCol: String = getScalarParam(timestampCol)
+
+  setDefault(timestampCol -> Left("timestamp"))
+
+  val connectionString = new ServiceParam[String](this, "connectionString", "Connection String " +
+    "for your storage account used for uploading files.")
+
+  def setConnectionString(v: String): this.type = setScalarParam(connectionString, v)
+
+  def getConnectionString: String = getScalarParam(connectionString)
+
+  val storageName = new ServiceParam[String](this, "storageName", "Storage Name " +
+    "for your storage account used for uploading files.")
+
+  def setStorageName(v: String): this.type = setScalarParam(storageName, v)
+
+  def getStorageName: String = getScalarParam(storageName)
+
+  val storageKey = new ServiceParam[String](this, "storageKey", "Storage Key " +
+    "for your storage account used for uploading files.")
+
+  def setStorageKey(v: String): this.type = setScalarParam(storageKey, v)
+
+  def getStorageKey: String = getScalarParam(storageKey)
+
+  val endpoint = new ServiceParam[String](this, "endpoint", "End Point " +
+    "for your storage account used for uploading files.")
+
+  def setEndpoint(v: String): this.type = setScalarParam(endpoint, v)
+
+  def getEndpoint: String = getScalarParam(endpoint)
+
+  val sasToken = new ServiceParam[String](this, "sasToken", "SAS Token " +
+    "for your storage account used for uploading files.")
+
+  def setSASToken(v: String): this.type = setScalarParam(sasToken, v)
+
+  def getSASToken: String = getScalarParam(sasToken)
+
+  val containerName = new ServiceParam[String](this, "containerName", "Container that will be " +
+    "used to upload files to.")
+
+  def setContainerName(v: String): this.type = setScalarParam(containerName, v)
+
+  def getContainerName: String = getScalarParam(containerName)
+
+  val sourceDF = new DataFrameParam(this, "sourceDF", "DataFrame as multivariate " +
```

Comment:
Shouldnt this be what is passed into fit? Or is this some other dataframe needed for the fitting?